scanner/config/config.go

```diff
@@ -1,9 +1,143 @@
 package config
 
+import (
+	"fmt"
+	"io"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/rs/zerolog"
+	"gopkg.in/yaml.v3"
+)
+
+var (
```

Comment:
Maybe we can have a function:
```
func NewDefaultConfiguration() Config {
    return Config{
        HTTPListenAddr: ":9443",
        GRPCListenAddr: ":8443",
        // ... other values ...
    }
}
```
And use it this way:
`cfg := NewDefaultConfiguration()`

That should make the default config immutable.

Reply:
I don't understand what you mean by "making the default config immutable". Can you please clarify?

Reply:
I think it's fine the way it is

Reply:
Let's hear from Yi what his concern is about then we will decide if its fine or not.

Reply:
Discussed with Victor, we can make the variable a package-level var. And only expose copies of the default, so modifications to it will not change the actual default variable.
We would still be able to change the default inadvertently inside the config package, but at least it will be contained so less risk.